src/overlays/actors/ovl_Bg_Ikana_Mirror/z_bg_ikana_mirror.c

```diff
@@ -28,99 +30,365 @@ const ActorInit Bg_Ikana_Mirror_InitVars = {
-    { { { 0.0f, 72.0f, 20.0f }, { 0.0f, 72.0f, 240.0f }, { 0.0f, 6.0f, 20.0f }, { 0.0f, 6.0f, 240.0f } } },
+static ColliderTrisInit sTrisInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER | AC_TYPE_OTHER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_TRIS,
+    },
+    9,
+    sTrisElementsInit,
 };
 
-// static ColliderQuadInit sQuadInit = {
-static ColliderQuadInit D_80B801FC = {
-    { 0x41, AT_NONE | AT_TYPE_OTHER, AC_NONE, OC1_NONE, OC2_FIRST_ONLY | OC2_UNK1 | OC2_HIT_PLAYER, 0x70, },
-    { 0xC1, { 0x41A00000, 0x41, 0xA0 }, { 0xC1CA6666, 0x41, 0xA0 }, TOUCH_ON | TOUCH_HIT | TOUCH_SFX_NORMAL | TOUCH_DREW_HITMARK, BUMP_NONE | BUMP_NO_DAMAGE | BUMP_NO_SWORD_SFX | BUMP_NO_HITMARK, OCELEM_NONE, },
-    { { { -1.8339854079840734e-09f, -1.8626939990440405e-09f, -1.9208972190654094e-09f }, { 221185.5625f, 0.0f, 0.003921568859368563f }, { 0.0f, 0.0f, 0.0f }, { 2.8698592549372254e-42f, 1.0761972206014595e-42f, 2.2420775429197073e-44f } } },
+static ColliderQuadInit sQuadInit[2] = {
+    {
+        {
+            COLTYPE_NONE,
+            AT_ON | AT_TYPE_OTHER,
+            AC_NONE,
+            OC1_NONE,
+            OC2_NONE,
+            COLSHAPE_QUAD,
+        },
+        {
+            ELEMTYPE_UNK0,
+            { 0x00200000, 0x00, 0x00 },
+            { 0x00000000, 0x00, 0x00 },
+            TOUCH_ON | TOUCH_SFX_NONE,
+            BUMP_NONE,
+            OCELEM_NONE,
+        },
+        { { { 0.0f, 72.0f, 20.0f }, { 0.0f, 72.0f, 240.0f }, { 0.0f, 6.0f, 20.0f }, { 0.0f, 6.0f, 240.0f } } },
+    },
+    {
+        {
+            COLTYPE_NONE,
+            AT_ON | AT_TYPE_OTHER,
+            AC_NONE,
+            OC1_NONE,
+            OC2_NONE,
+            COLSHAPE_QUAD,
+        },
+        {
+            ELEMTYPE_UNK0,
+            { 0x00200000, 0x00, 0x00 },
+            { 0x00000000, 0x00, 0x00 },
+            TOUCH_ON | TOUCH_SFX_NONE,
+            BUMP_NONE,
+            OCELEM_NONE,
+        },
+        { { { 25.3f, 20.0f, 20.0f }, { 25.3f, 20.0f, 240.0f }, { -25.3f, 20.0f, 20.0f }, { -25.3f, 20.0f, 240.0f } } },
+    }
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B8021C[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 220, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 200, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgIkanaMirror_SetupWait(BgIkanaMirror* this);
+void BgIkanaMirror_Wait(BgIkanaMirror* this, PlayState* play);
+void BgIkanaMirror_SetupEmitLight(BgIkanaMirror* this);
+void BgIkanaMirror_EmitLight(BgIkanaMirror* this, PlayState* play);
+
+void BgIkanaMirror_SetQuadVertices(BgIkanaMirror* this) {
+    ColliderQuadDimInit* dim;
+    ColliderQuad* collider;
+    Vec3f a;
+    Vec3f b;
+    Vec3f c;
+    Vec3f d;
```

Comment:
Maybe `v0`, `v1`, etc. I know that they're just used as points but it's a bit clearer to make them look more similar.

Reply:
Renamed them to v0, v1, v2, v3. Tried making them into an array, but had to reverse the accesses from 3 to 0 to match and I didn't like that.

Reply:
Will be pushed in next commit.